lib/src/image/ops/resize_with_crop_or_pad_op.dart

```diff
@@ -82,6 +100,37 @@ class ResizeWithCropOrPadOp implements ImageOperator {
     return image;
   }
 
+  Tuple2<int, int> _computeCropPosition(int targetSize, int imageSize,
+      [int cropPosition]) {
+    int srcLT;
+    int srcRB;
+
+    if (cropPosition != null) {
+      srcLT = cropPosition; // custom crop
+    } else {
+      srcLT = (imageSize - targetSize) ~/ 2; // centered crop
+    }
+    srcRB = srcLT + targetSize;
+
+    return Tuple2<int, int>(srcLT, srcRB);
+  }
+
+  // This function is used to check the crop custom crop position is valid
+  void _checkCropPositionArgument(
+      int cropWidthCustomPosition, int cropHeightCustomPosition, int w, int h) {
+    // Check crop position inpute if provided
+    if (cropWidthCustomPosition != null && cropHeightCustomPosition != null) {
+      // Return an error if the crop position is outside of the image
+      if ((cropWidthCustomPosition + _targetWidth > w) ||
+          (cropHeightCustomPosition + _targetHeight > h)) {
+        int leftWidth = cropWidthCustomPosition + _targetWidth;
+        int bottomHeight = cropHeightCustomPosition + _targetHeight;
+        throw ArgumentError(
+            "The crop position is outside the image : crop(x:x+cropWidth,y+cropHeight) = ($cropWidthCustomPosition:$leftWidth, $cropHeightCustomPosition:$bottomHeight) not in imageSize(x,y) = ($w, $h)");
+      }
+    }
+  }
```

Comment:
We can then replace it with, 

```
void _checkCropPositionArgument(int w, int h)
```

and use class variables `_cropLeft` and `_cropRight`

Reply:
Obviously!